contribs/gnodev/cmd/gnobro/main.go

```diff
@@ -0,0 +1,452 @@
+	"github.com/charmbracelet/lipgloss"
+	charmlog "github.com/charmbracelet/log"
+	"github.com/charmbracelet/ssh"
+	"github.com/charmbracelet/wish"
+	"github.com/charmbracelet/wish/activeterm"
+	"github.com/charmbracelet/wish/bubbletea"
+	"github.com/charmbracelet/wish/logging"
+
+	"github.com/gnolang/gno/contribs/gnodev/pkg/browser"
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gno.land/pkg/integration"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+const gnoPrefix = "gno.land"
+
+type broCfg struct {
+	readonly       bool
+	remote         string
+	devEndpoint    string
+	chainID        string
+	defaultAccount string
+	defaultRealm   string
+	sshListener    string
+	sshHostKeyPath string
+	banner         bool
+}
+
+var defaultBroOptions = broCfg{
+	remote:         "127.0.0.1:26657",
+	devEndpoint:    "",
+	sshListener:    "",
+	defaultRealm:   "gno.land/r/gnoland/home",
+	chainID:        "dev",
+	sshHostKeyPath: ".ssh/id_ed25519",
+}
+
+func main() {
+	cfg := &broCfg{}
+
+	stdio := commands.NewDefaultIO()
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "gnobro",
+			ShortUsage: "gnobro [flags] [pkg_path]",
+			ShortHelp:  "Gno Browser, a realm explorer",
+			LongHelp: `Gnobro is a terminal user interface (TUI) that allows you to browse realms within your
+terminal. It automatically connects to Gnodev for real-time development. In
+addition to hot reload, it also has the ability to execute commands and interact
+with your realm.
+`,
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execBrowser(cfg, args, stdio)
+		})
+
+	cmd.Execute(context.Background(), os.Args[1:])
+}
+
+func (c *broCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.remote,
+		"target",
+		defaultBroOptions.remote,
+		"target gnoland address",
+	)
+
+	fs.StringVar(
+		&c.chainID,
+		"chainid",
+		defaultBroOptions.chainID,
+		"chainid",
+	)
+
+	fs.StringVar(
+		&c.defaultAccount,
+		"account",
+		defaultBroOptions.defaultAccount,
+		"default local account to use",
+	)
+
+	fs.StringVar(
+		&c.defaultRealm,
+		"default-realm",
+		defaultBroOptions.defaultRealm,
+		"default realm to display when gnobro start and no argument are provided",
+	)
+
+	fs.StringVar(
+		&c.sshListener,
+		"ssh",
+		defaultBroOptions.sshListener,
+		"ssh server listener address",
+	)
+
+	fs.StringVar(
+		&c.sshHostKeyPath,
+		"ssh-key",
+		defaultBroOptions.sshHostKeyPath,
+		"ssh host key path",
+	)
+
+	fs.StringVar(
+		&c.devEndpoint,
+		"dev",
+		defaultBroOptions.devEndpoint,
+		"dev endpoint, if empty will default to `ws://<target>:8888`",
+	)
+
+	fs.BoolVar(
+		&c.banner,
+		"banner",
+		defaultBroOptions.banner,
+		"if enable, display a banner",
```

Comment:
Typo here, should be `enabled`

Reply:
93f0742c423aae64eeeadd3c386c6262a3947592